parser/parser.go

```diff
@@ -27,23 +28,32 @@ type QueryField struct {
 
 type QueryFieldList []QueryField
 
-func ParseQueryDir(dir string, schema *ast.Schema) (QueryFieldList, error) {
-	fields := QueryFieldList{}
+func ParseQuerySource(source string, schema *ast.Schema) (QueryFieldList, error) {
+	//check if file or directory
+	fileInfo, err := os.Stat(source)
+	if err != nil {
+		return nil, err
+	}
+
+	if fileInfo.IsDir() {
```

Comment:
wdyt if instead of having a comment explain what we intend the code to do, we can wrap this into a function and do this?

```go
if ok, err := isDirectory(source); ok {`
```